src/export/StructureDefinitionExporter.ts

```diff
@@ -61,16 +78,244 @@ export class StructureDefinitionExporter implements Fishable {
   ) {}
 
+  /**
+   * Processes the fshDefinition's parent, validating it according to its type.
+   * Returns the parent's StructureDefinition as the basis for the new StructureDefinition
+   * for the provided fshDefinition.
+   * @param {Extension | Profile | Logical | Resource} fshDefinition - The definition
+   *        to be preprocessed. It is updated directly based on this processing.
+   * @returns {StructureDefinition} for this fshDefinition
+   * @private
+   */
+  private getStructureDefinition(
+    fshDefinition: Profile | Extension | Logical | Resource
+  ): StructureDefinition {
+    // Process/validate the fshDefinition.parent value with the purpose of
+    // obtaining the parent's StructureDefinition as the basis for this
+    // fshDefinition's StructureDefinition.
+    // RECALL: fshDefinition.parent can be specified as the 'id', 'name', or 'url'.
+
+    if (isEmpty(fshDefinition.parent)) {
+      // Handle cases where the parent is not specified by throwing an error.
+      // - Profile has a hard requirement to define a parent and does not have a default
+      //   parent defined in the Profile class. Therefore, we need to throw this error.
+      // - Extension, Logical, and Resource all have a default parent, so we should never
+      //   get to this point where it is not defined. That said, if something somewhere
+      //   in the code were to change/remove the default, this error will then be thrown.
+      throw new ParentNotProvidedError(fshDefinition.name, fshDefinition.sourceInfo);
+    }
+
+    let defnType: Type;
+    if (fshDefinition instanceof Extension) {
+      defnType = Type.Extension;
+    } else if (fshDefinition instanceof Logical) {
+      defnType = Type.Logical;
+    } else {
+      defnType = Type.Resource;
+    }
+
+    if (fshDefinition.name === fshDefinition.parent) {
+      const result = this.fishForMetadata(fshDefinition.parent, defnType);
+      throw new ParentDeclaredAsNameError(
+        fshDefinition.constructorName,
+        fshDefinition.name,
+        fshDefinition.sourceInfo,
+        result?.url
+      );
+    }
+
+    if (fshDefinition.id === fshDefinition.parent) {
+      const result = this.fishForMetadata(fshDefinition.parent, defnType);
+      throw new ParentDeclaredAsIdError(
+        fshDefinition.constructorName,
+        fshDefinition.name,
+        fshDefinition.id,
+        fshDefinition.sourceInfo,
+        result?.url
+      );
+    }
+
+    // Now that we have a usable fshDefinition.parent, retrieve its StructureDefinition.
+    // Then make sure it is a valid StructureDefinition based on the type of the fshDefinition.
+
+    let parentJson = this.fishForFHIR(fshDefinition.parent);
+    if (
+      !parentJson &&
+      fshDefinition instanceof Logical &&
+      (fshDefinition.parent === 'Base' ||
+        fshDefinition.parent === 'http://hl7.org/fhir/StructureDefinition/Base')
+    ) {
+      parentJson = this.createR4BaseStructureDefinition();
+    }
```

Comment:
Good point about R5. This specific logic is to create a non-existent Base StructureDefinition for R4. 

Base will exist if R5 and beyond. I looked at FHIRDefinition, and since R5 Base is `"kind": "complex-type"`, it will be added to the "types" map which will not really help here. I need to think this through, so I'm going to hold off making any changes until I can walk this through the code and come up with a proposal for you.

Reply:
OK.  Sounds good.  I commented elsewhere that I wondered if we should allow complex types as parents of logical models too.  This finding (the `Base` is a `complex-type`) would seem to suggest that perhaps we should.

Reply:
@cmoesel @ngfreiter The current commits of this PR now have support for complex types. That pretty much solved this issue since we now will fish for complex-types for parents of logical models. I did some testing with both R4 and R5 in my .fhir cache. When I set the fhirVersion in my sushi-config.yaml to 4.0.1, my logical model used the "hard-coded" 4.0.1 version of `Base` and when I set it to 4.6.0, it used the R5 version of `Base` as it should have. So, there was nothing to change to properly handle the use of the R4 vs the R5 `Base` parent.